src/combinatorics/set_family/shadow.lean

```diff
@@ -121,4 +127,113 @@ begin
     refl }
 end
 
+end shadow
+
+open_locale finset_family
+
+section up_shadow
+variables [decidable_eq α] [fintype α] {𝒜 : finset (finset α)} {s t : finset α} {a : α} {k : ℕ}
+
+/-- The upper shadow of a set family `𝒜` is all sets we can get by adding one element to any set in
+`𝒜`, and the (`k` times) iterated up_shadow (`up_shadow^[k]`) is all sets we can get by removing `k`
```

Comment:
Copy-paste error here? The iterated up shadow is from adding `k` elements

Reply:
Well spotted.